js/ui/components/clubhouse.js

```diff
@@ -200,6 +206,62 @@ var ClubhouseNotificationBanner = new Lang.Class({
         }
     },
 
+    _rearrange_elements: function() {
+        let contentBox = this._bodyStack.get_parent();
+        contentBox.add_style_class_name('clubhouse-content-box');
+        let hbox = contentBox.get_parent();
+        let vbox = hbox.get_parent();
+
+        let wrapBin = new St.Bin({ x_fill: true,
+                                   y_fill: true,
+                                   x_align: St.Align.END,
+                                   y_align: St.Align.END,
+                                   x_expand: true,
+                                   y_expand: true });
+        hbox.add_child(wrapBin);
+
+        // A Clutter.BinLayout is used to rearrange the notification
+        // elements in layers:
+        let wrapWidget = new St.Widget({ layout_manager: new Clutter.BinLayout(),
+                                         x_align: Clutter.ActorAlign.FILL,
+                                         y_align: Clutter.ActorAlign.END });
+        wrapBin.set_child(wrapWidget);
+
+        this._iconBin.set_fill(false, false);
+        this._iconBin.set_x_expand(true);
+        this._iconBin.set_y_expand(true);
+        this._iconBin.set_x_align(Clutter.ActorAlign.END);
+        this._iconBin.set_y_align(Clutter.ActorAlign.END);
```

Comment:
Is it possible to use `Object.assign()` to simplify this code and the one below?

Reply:
I didn't knew about `Object.assign()`. Yes using it the code is more readable and consistent with the parameters passed to the constructors.